addons/binding/org.openhab.binding.knx/src/main/java/org/openhab/binding/knx/internal/channel/AbstractSpec.java

```diff
@@ -0,0 +1,50 @@
+/**
+ * Copyright (c) 2010-2017 by the respective copyright holders.
+ *
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ */
+package org.openhab.binding.knx.internal.channel;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.eclipse.jdt.annotation.Nullable;
+
+import tuwien.auto.calimero.GroupAddress;
+import tuwien.auto.calimero.exception.KNXFormatException;
+
+/**
+ * Base class for telegram meta-data
+ *
+ * @author Simon Kaufmann - initial contribution and API.
+ *
+ */
+@NonNullByDefault
+public abstract class AbstractSpec {
```

Comment:
@sjka With the advanced binding in mind, this class could be a regular class if you add a List<GroupAddress> getAddresses() method, which - depending on the use case - contain only one or more GroupAddresses. That way there is no need for a CommandSpec, ListenSpec and so forth. Consequently, in KNXChannelType the getCommandSpec(...) and similar methods can be simplified. In KNXClient.writeToKNX(...) you could then take the first (and only) GroupAddress returned by getGroupAddresses() to make the sendToKNX(...) call